src/math/PercentageMath.sol

```diff
@@ -40,16 +40,16 @@ library PercentageMath {
         }
     }
 
-    /// @notice Executes a percentage subtraction (x * (1 - p)), rounded up.
+    /// @notice Executes a percentage subtraction (x * (1 - p)), rounded half up.
     /// @param x The value to which to subtract the percentage.
-    /// @param percentage The percentage of the value to subtract.
+    /// @param percentage The percentage of the value to subtract (in bps).
     /// @return y The result of the subtraction.
     function percentSub(uint256 x, uint256 percentage) internal pure returns (uint256 y) {
-        // Must revert if
-        // percentage > PERCENTAGE_FACTOR
-        //     or x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max
-        // <=> percentage > PERCENTAGE_FACTOR
-        //     or ((PERCENTAGE_FACTOR - percentage) > 0 and x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage))
+        // Underflow if
+        //     percentage > PERCENTAGE_FACTOR
+        // Overflow if
+        //     x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max
+        // <=> (PERCENTAGE_FACTOR - percentage) > 0 and x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)
```

Comment:
The <=> no longer works here I think

Reply:
It would be equivalent if we add `Let percentage <= PERCENTAGE_FACTOR`:

1. 
`(PERCENTAGE_FACTOR - percentage) > 0 and x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)` implies that `x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max`

2. `x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max` implies that either:
- `PERCENTAGE_FACTOR - percentage = 0 and HALF_PERCENTAGE_FACTOR > type(uint256).max`: impossible; OR
- `PERCENTAGE_FACTOR - percentage > 0 and x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)`

Correct?

Reply:
Yes in this case it works but it overcomplicates things IMO.

Here I read:
(Underflow if A) and (overflow if B) <=> C and D
But I should read:
(Underflow if A) and (overflow if (B <=> C))

That's why I did:
revert if (A or B <=> C or D) which looks better IMO

Reply:
Understood & agreed! Submitted a new suggestion